src/Middleware/SocialAuthMiddleware.php

```diff
@@ -175,18 +175,15 @@ protected function _getUser($providerName, ServerRequest $request)
 
                 return false;
             }
-        } elseif ($identity->email) {
-            $user = $this->_userModel->find($finder)
-                ->where([
-                    $this->_userModel->aliasField($this->config('fields.email')) => $identity->email
-                ])
-                ->first();
         }
 
         $profile = $this->_getProfileEntity($providerName, $identity, $accessToken, $profile ?: null);
+        if ($profile->isDirty()) {
+            $this->_saveProfile($profile);
```

Comment:
Why save here? The new user callback could actually check if it's a new profile entity ($profile->isNew()) to decide whether it needs to find a corresponding user entity or create a new one.